### DISCUSSION TITLE
Modal: return focus to trigger without scrolling?

### DISCUSSION THREAD

**Discussion by tinymachine** [incluud/accessible-astro-components#15] [Ideas]
Hi @markteekman — right now when you close a Modal, focus returns to the trigger (which is great). But without the `preventScroll` option on the `focus()` method, the element scrolls fully into view if it's partially scrolled out of the viewport. I found this scrolling behavior unexpected. Just a thought that you might consider changing this line...

https://github.com/markteekman/accessible-astro-components/blob/7ff14fee3d320247c890ec04ea58aaba61407c1f/Modal.astro#L78

to

```js
modalTrigger.focus({ preventScroll: true })` 
```

(If I'm reading it correctly, caniuse.com [reports](https://caniuse.com/mdn-api_htmlelement_focus_options_preventscroll_parameter) that this option isn't supported on Safari 15.4, but it's working fine for me on macOS and iOS. 🤷🏾‍♂️)

**Comment by markteekman** [comment]
Hey @tinymachine, thanks for looking so thoroughly at these components, really appreciate the time you're taking in helping to improve them 🙂 

That's a nice use case you've found! And definitely some unwanted behavior. Your solution to this sounds like the logical way to go, thanks for looking into it! I'll create an issue and when I've got the time I will implement it 👍🏼
